### PR TITLE
WIP: Raffle sample in Java, #25485

### PR DESCRIPTION
**Pull request by patriknw**
Refs #25485

### CODE REVIEW

File: akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      @Override
+      public Raffle emptyState() {
+        return new Raffle();
+      }
+
+      @Override
+      public CommandHandler<Command, Event, Raffle> commandHandler() {
+        CommandHandlerBuilder<Command, Event, Raffle, Raffle> notStartedHandler =
+            commandHandlerBuilder(raffle -> raffle.getState() == RaffleState.NOT_STARTED)
+                .matchCommand(StartRaffle.class, (raffle, cmd) -> Effect().persist(new RaffleStarted()));
```
Comment (renatocaval) [inline akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java]: Would it make sense to have convenience builder methods when using `null` state?

```java
public Raffle emptyState() { return null; }

CommandHandlerBuilder<Command, Event, Raffle, Raffle> notStartedHandler =
  
  commandHandlerBuilderNull()
    .matchCommand(StartRaffle.class, (cmd) -> Effect().persist(new RaffleStarted()));
...
}
```

Comment (renatocaval) [reply akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java]: Btw, the name `commandHandlerBuilderNull` is horrible. I just want to drop the idea.

Comment (renatocaval) [reply akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java]: Hmm, just saw that there a variant that accepts a Predicate and that we can write:

`commandHandlerBuilder(Objects::isNull)`

File: akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+  // TODO replies are not implemented in this example yet
+
+  private enum RaffleState {
+    NOT_STARTED,
+    RUNNING,
+    COMPLETED
+  }
+
+  // mutable domain model impl, completely free from Akka stuff
```
Comment (jboner) [inline akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java]: I like this approach. Both the separation of the domain model and the Akka specific stuff, and the mutable approach, which feels more idiomatic Java.

File: akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java (hunk 1 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java]: there will be a match command with Function<Command> as alternative to Function<State, Command>, I think I added that in some other PR

**Comment by TimMoore** [comment]
It would be good to show:

* A query command that replies with the current state of the raffle
* Acknowledgement of other commands

**Comment by patriknw** [comment]
This example is a good addition, but it's pending the support of mutable state 
https://github.com/akka/akka/issues/25740

Keeping the PR as WIP

**Comment by patriknw** [comment]
Might pick this up again as a blog post example.

**Pull request closed without merging**